cve-2022-21907/cve-2022-21907.audit

```diff
@@ -0,0 +1,102 @@
+      reference   : "CVE|CVE-2022-21907"
+      see_also    : "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2022-21907"
+      value_type  : POLICY_DWORD
+      value_data  : 0
+      reg_key     : "HKLM\System\Currentcontrolset\Services\Http\Parameters"
+      reg_item    : "EnableTrailerSupport"
+      reg_option  : CAN_BE_NULL
+    </custom_item>
+  </then>
+
+  <else>
+    <if>
+      <condition type:"AND">
+        <custom_item>
+          type        : REGISTRY_SETTING
+          description : "Windows 10 is installed"
+          value_type  : POLICY_TEXT
+          value_data  : "^[Ww][Ii][Nn][Dd][Oo][Ww][Ss] 10.+$"
+          reg_key     : "HKLM\Software\Microsoft\Windows Nt\Currentversion"
+          reg_item    : "ProductName"
+          check_type  : CHECK_REGEX
+        </custom_item>
+
+        <custom_item>
+          type        : REGISTRY_SETTING
+          description : "Check Windows Release"
+          value_type  : POLICY_TEXT
+          value_data  : "1809"
+          reg_key     : "HKLM\Software\Microsoft\Windows Nt\Currentversion"
+          reg_item    : "ReleaseId"
+          check_type  : CHECK_REGEX
+        </custom_item>
+      </condition>
+
+      <then>
+        <custom_item>
+          type        : REGISTRY_SETTING
+          description : "CVE-2022-21907 - HTTP Protocol Stack Remote Code Execution Vulnerability"
+          info        : "According to the latest announcement issued by the Microsoft Security Response Center, Microsoft has issued a patch for a high-severity vulnerability found in Windows Server 2019 and Windows 10 in the latest cumulative update. The vulnerability ID is CVE-2022-21907, and the vulnerability can be exploited by sending specially crafted packets to exploit the HTTP protocol stack to launch an attack. Enterprise Administrators can also use the registry to disable certain inactive features to improve security if the lates cumulative update cannot be installed in a timely manner to fix the vulnerability. This mitigation applies to Windows Server version 2019 and Windows 10 (1809) and specifically disables the Trailer interface via the registry, which is used for HTTP header functionality."
+          reference   : "CVE|CVE-2022-21907"
+          see_also    : "https://msrc.microsoft.com/update-guide/vulnerability/CVE-2022-21907"
+          value_type  : POLICY_DWORD
+          value_data  : 0
+          reg_key     : "HKLM\System\Currentcontrolset\Services\Http\Parameters"
+          reg_item    : "EnableTrailerSupport"
+          reg_option  : CAN_BE_NULL
+        </custom_item>
+      </then>
+
+      <else>
+        <report type:"PASSED">
```

Comment:
may want to change this to a WARNING so people don't see it and think they are not vulnerable.

Reply:
done